cf/polyfills.js

```diff
@@ -13,61 +13,69 @@ const IPv4Reg = new RegExp(`^${v4Str}$`)
 const v6Seg = '(?:[0-9a-fA-F]{1,4})'
 const IPv6Reg = new RegExp(
   '^(' +
-    `(?:${v6Seg}:){7}(?:${v6Seg}|:)|` +
-    `(?:${v6Seg}:){6}(?:${v4Str}|:${v6Seg}|:)|` +
-    `(?:${v6Seg}:){5}(?::${v4Str}|(:${v6Seg}){1,2}|:)|` +
-    `(?:${v6Seg}:){4}(?:(:${v6Seg}){0,1}:${v4Str}|(:${v6Seg}){1,3}|:)|` +
-    `(?:${v6Seg}:){3}(?:(:${v6Seg}){0,2}:${v4Str}|(:${v6Seg}){1,4}|:)|` +
-    `(?:${v6Seg}:){2}(?:(:${v6Seg}){0,3}:${v4Str}|(:${v6Seg}){1,5}|:)|` +
-    `(?:${v6Seg}:){1}(?:(:${v6Seg}){0,4}:${v4Str}|(:${v6Seg}){1,6}|:)|` +
-    `(?::((?::${v6Seg}){0,5}:${v4Str}|(?::${v6Seg}){1,7}|:))` +
-    ')(%[0-9a-zA-Z-.:]{1,})?$'
+  `(?:${v6Seg}:){7}(?:${v6Seg}|:)|` +
+  `(?:${v6Seg}:){6}(?:${v4Str}|:${v6Seg}|:)|` +
+  `(?:${v6Seg}:){5}(?::${v4Str}|(:${v6Seg}){1,2}|:)|` +
+  `(?:${v6Seg}:){4}(?:(:${v6Seg}){0,1}:${v4Str}|(:${v6Seg}){1,3}|:)|` +
+  `(?:${v6Seg}:){3}(?:(:${v6Seg}){0,2}:${v4Str}|(:${v6Seg}){1,4}|:)|` +
+  `(?:${v6Seg}:){2}(?:(:${v6Seg}){0,3}:${v4Str}|(:${v6Seg}){1,5}|:)|` +
+  `(?:${v6Seg}:){1}(?:(:${v6Seg}){0,4}:${v4Str}|(:${v6Seg}){1,6}|:)|` +
+  `(?::((?::${v6Seg}){0,5}:${v4Str}|(?::${v6Seg}){1,7}|:))` +
+  ')(%[0-9a-zA-Z-.:]{1,})?$'
 )
 
 const textEncoder = new TextEncoder()
 export const crypto = {
-  randomBytes: l => Crypto.getRandomValues(Buffer.alloc(l)),
-  pbkdf2Sync: async(password, salt, iterations, keylen) => Crypto.subtle.deriveBits(
-    {
-      name: 'PBKDF2',
-      hash: 'SHA-256',
-      salt,
-      iterations
-    },
-    await Crypto.subtle.importKey(
-      'raw',
-      textEncoder.encode(password),
-      'PBKDF2',
-      false,
+  randomBytes: (l) => Crypto.getRandomValues(Buffer.alloc(l)),
+  pbkdf2Sync: async (password, salt, iterations, keylen) =>
+    Crypto.subtle.deriveBits(
+      {
+        name: 'PBKDF2',
+        hash: 'SHA-256',
+        salt,
+        iterations
+      },
+      await Crypto.subtle.importKey(
+        'raw',
+        textEncoder.encode(password),
+        'PBKDF2',
+        false,
+        ['deriveBits']
+      ),
+      keylen * 8,
       ['deriveBits']
     ),
-    keylen * 8,
-    ['deriveBits']
-  ),
   createHash: (type) => ({
     update: (x) => ({
       digest: () => {
-        return type === 'sha256'
-          ? Crypto.subtle.digest('SHA-256', x)
-          : Crypto.subtle.digest('MD5', x)
+        if (type !== 'sha256')
+          throw Error('createHash only supports sha256 on cloudflare.')
```

Comment:
This won't just run on Cloudflare. I would suggest a more generic error message, like `createHash only supports sha256 in this environment.`